linker missing libsecp256k1

### Issue and Steps to Reproduce

Here we are back to #4229 (and the whole tree mentioned in there).

This whole issue is again caused by outdated git submodules.

Oroginally I wrote following:

When compiling the latest `master` I get following on systems which do not have the current `libsecp256k1` installed in `/usr/local`, but rather have merely the (rather _old_) Ubuntu package ([`libsecp256k1-dev 0.1~20170810-2`](https://packages.ubuntu.com/search?suite=groovy&section=all&arch=any&keywords=libsecp256k1&searchon=names)).

```
~/src/lightning$ git describe
v0.9.2-142-g77478408f
~/src/lightning$ ./configure
...
~/src/lightning$ make
...
/usr/bin/ld: common/node_id.o: in function `pubkey32_from_node_id':
/home/nisim/src/lightning/common/node_id.c:34: undefined reference to `secp256k1_xonly_pubkey_from_pubkey'
/usr/bin/ld: bitcoin/pubkey.o: in function `fromwire_pubkey32':
/home/nisim/src/lightning/bitcoin/pubkey.c:136: undefined reference to `secp256k1_xonly_pubkey_parse'
/usr/bin/ld: bitcoin/pubkey.o: in function `towire_pubkey32':
/home/nisim/src/lightning/bitcoin/pubkey.c:148: undefined reference to `secp256k1_xonly_pubkey_serialize'
/usr/bin/ld: bitcoin/pubkey.o: in function `pubkey32_to_hexstr':
/home/nisim/src/lightning/bitcoin/pubkey.c:157: undefined reference to `secp256k1_xonly_pubkey_serialize'
collect2: error: ld returned 1 exit status
make: *** [Makefile:531: lightningd/lightning_hsmd] Error 1
```

Would it be possible to statically link against any of the secp256k1 which are around?

```
~/src/lightning$ find . -type d -name '*secp256k1*'
./external/x86_64-linux-gnu/libwally-core-build/src/secp256k1
./external/libwally-core/src/secp256k1
./.git/modules/external/libwally-core/modules/src/secp256k1
```

@cdecker I have a current issue with dc38e65ed. Try to start with a fresh clone of `lightning` and make sure there is no `secp256k1_recovery.h` in `/usr/include` or `/usr/local/include`. After running `./configure`, do a parallel make: `make -j4` and you should see this:

```
In file included from ./common/iso4217.h:4,
                 from common/iso4217.c:3:
./wire/wire.h:7:10: fatal error: secp256k1_recovery.h: No such file or directory
    7 | #include <secp256k1_recovery.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

It does not happen when `make` is run again, meaning when the submodules are already initialized. Because the compiler looks for include files in `external/libwally-core/src/secp256k1/include` but during a first-run parallel make it is not initialized yet… Does anyone use `make -jX` where `X` is more than `1`?

--- a/common/Makefile
+++ b/common/Makefile
@@ -45,6 +45,7 @@ COMMON_SRC_NOGEN :=				\
 	common/initial_channel.c		\
 	common/initial_commit_tx.c		\
 	common/io_lock.c			\
+	common/iso4217.c			\
 	common/json.c				\
 	common/json_helpers.c			\
 	common/json_stream.c			\
@@ -100,7 +101,6 @@ COMMON_HEADERS_NOGEN := $(COMMON_SRC_NOGEN:.c=.h)	\
 	common/errcode.h				\
 	common/gossip_constants.h			\
 	common/htlc.h					\
-	common/iso4217.h				\
 	common/json_command.h				\
 	common/jsonrpc_errors.h				\
 	common/overflows.h				\

--- a/wire/Makefile
+++ b/wire/Makefile
@@ -79,13 +79,13 @@ wire/bolt12_wire.csv.raw: bolt-precheck
 	@$(BOLT_EXTRACT) $(LOCAL_BOLTDIR)/12*.md > $@
 
 # How to make the exp versions from non-exp versions:
-wire/peer_exp_wire.csv: wire/peer_wire.csv
+wire/peer_exp_wire.csv: wire/peer_wire.csv $(PEER_EXP_PATCHES)
 	@set -e; T=$@; trap "rm -f $$T.$$$$" 0; cp $< $$T.$$$$; for p in $(PEER_EXP_PATCHES); do echo APPLY $$p; patch $$T.$$$$ $$p; done; mv $$T.$$$$ $$T
 
-wire/onion_exp_wire.csv: wire/onion_wire.csv
+wire/onion_exp_wire.csv: wire/onion_wire.csv $(ONION_EXP_PATCHES)
 	@set -e; T=$@; trap "rm -f $$T.$$$$" 0; cp $< $$T.$$$$; for p in $(ONION_EXP_PATCHES); do echo APPLY $$p; patch $$T.$$$$ $$p; done; mv $$T.$$$$ $$T
 
-wire/bolt12_exp_wire.csv: wire/bolt12_wire.csv
+wire/bolt12_exp_wire.csv: wire/bolt12_wire.csv $(BOLT12_EXP_PATCHES)
 	@set -e; T=$@; trap "rm -f $$T.$$$$" 0; cp $< $$T.$$$$; for p in $(BOLT12_EXP_PATCHES); do echo APPLY $$p; patch $$T.$$$$ $$p; done; mv $$T.$$$$ $$T
 
 # These can be deleted.

--- a/wire/bolt12_wiregen.c
+++ b/wire/bolt12_wiregen.c
@@ -1562,4 +1562,4 @@ bool invoice_error_is_valid(const struct tlv_invoice_error *record, size_t *err_
 	return tlv_fields_valid(record->fields, err_index);
 }
 
-// SHA256STAMP:b09ebf868d0b5b9792d0054bee17d763c1243d18c419847c554c63aefb7a8548
+// SHA256STAMP:e82f2f1c0f73838262e59ffeb3572d5e8ce6efcca7c8ae3c36f6448adacde16b

--- a/wire/bolt12_wiregen.h
+++ b/wire/bolt12_wiregen.h
@@ -316,4 +316,4 @@ struct fallback_address *fromwire_fallback_address(const tal_t *ctx, const u8 **
 
 
 #endif /* LIGHTNING_WIRE_BOLT12_WIREGEN_H */
-// SHA256STAMP:b09ebf868d0b5b9792d0054bee17d763c1243d18c419847c554c63aefb7a8548
+// SHA256STAMP:e82f2f1c0f73838262e59ffeb3572d5e8ce6efcca7c8ae3c36f6448adacde16b

--- a/wire/common_wiregen.c
+++ b/wire/common_wiregen.c
@@ -100,4 +100,4 @@ bool fromwire_custommsg_out(const tal_t *ctx, const void *p, u8 **msg)
 	fromwire_u8_array(&cursor, &plen, *msg, msg_len);
 	return cursor != NULL;
 }
-// SHA256STAMP:304c3015d0df2a9436adf3c2c96365d438781a5cf55b9ccba73a05dd7bc28a4d
+// SHA256STAMP:65c5b5accba767d72ee72ac8915a4fcf0524ef688d3bdfaa1e4a548ba041b354

--- a/wire/common_wiregen.h
+++ b/wire/common_wiregen.h
@@ -41,4 +41,4 @@ bool fromwire_custommsg_out(const tal_t *ctx, const void *p, u8 **msg);
 
 
 #endif /* LIGHTNING_WIRE_COMMON_WIREGEN_H */
-// SHA256STAMP:304c3015d0df2a9436adf3c2c96365d438781a5cf55b9ccba73a05dd7bc28a4d
+// SHA256STAMP:65c5b5accba767d72ee72ac8915a4fcf0524ef688d3bdfaa1e4a548ba041b354

--- a/wire/onion_printgen.c
+++ b/wire/onion_printgen.c
@@ -859,4 +859,4 @@ void printonion_wire_tlv_message(const char *tlv_name, const u8 *msg) {
 		printwire_tlvs(tlv_name, &msg, &plen, print_tlvs_encmsg_tlvs, ARRAY_SIZE(print_tlvs_encmsg_tlvs));
 	}
 }
-// SHA256STAMP:897accc52196e69cffeee8386a5e1af8754fe49baefd58460c788024e4e8dd6f
+// SHA256STAMP:a6e1f0cd5e6b270cf9269569f56b283241f4b5c7f393ef86423399b51f1f6b98

--- a/wire/onion_printgen.h
+++ b/wire/onion_printgen.h
@@ -58,4 +58,4 @@ void printwire_mpp_timeout(const char *fieldname, const u8 *cursor);
 
 void printwire_onionmsg_path(const char *fieldname, const u8 **cursor, size_t *plen);
 #endif /* LIGHTNING_WIRE_ONION_PRINTGEN_H */
-// SHA256STAMP:897accc52196e69cffeee8386a5e1af8754fe49baefd58460c788024e4e8dd6f
+// SHA256STAMP:a6e1f0cd5e6b270cf9269569f56b283241f4b5c7f393ef86423399b51f1f6b98

--- a/wire/onion_wiregen.c
+++ b/wire/onion_wiregen.c
@@ -1026,4 +1026,4 @@ bool fromwire_mpp_timeout(const void *p)
 		return false;
 	return cursor != NULL;
 }
-// SHA256STAMP:897accc52196e69cffeee8386a5e1af8754fe49baefd58460c788024e4e8dd6f
+// SHA256STAMP:720f5585d10f973a398e821541c6a2f0841c338c0c73f53bd4215d7bcad378d1

--- a/wire/onion_wiregen.c
+++ b/wire/onion_wiregen.c
@@ -1026,4 +1026,4 @@ bool fromwire_mpp_timeout(const void *p)
 		return false;
 	return cursor != NULL;
 }
-// SHA256STAMP:720f5585d10f973a398e821541c6a2f0841c338c0c73f53bd4215d7bcad378d1
+// SHA256STAMP:a6e1f0cd5e6b270cf9269569f56b283241f4b5c7f393ef86423399b51f1f6b98

--- a/wire/onion_wiregen.h
+++ b/wire/onion_wiregen.h
@@ -317,4 +317,4 @@ bool fromwire_mpp_timeout(const void *p);
 
 
 #endif /* LIGHTNING_WIRE_ONION_WIREGEN_H */
-// SHA256STAMP:897accc52196e69cffeee8386a5e1af8754fe49baefd58460c788024e4e8dd6f
+// SHA256STAMP:a6e1f0cd5e6b270cf9269569f56b283241f4b5c7f393ef86423399b51f1f6b98

--- a/wire/peer_printgen.c
+++ b/wire/peer_printgen.c
@@ -2130,4 +2130,4 @@ void printpeer_wire_tlv_message(const char *tlv_name, const u8 *msg) {
 		printwire_tlvs(tlv_name, &msg, &plen, print_tlvs_onion_message_tlvs, ARRAY_SIZE(print_tlvs_onion_message_tlvs));
 	}
 }
-// SHA256STAMP:b3a92a714208711191936bbdb7ff0f0a2ca412941ea95e67210029e5249f3d60
+// SHA256STAMP:1b0c5319cd9ab8c0281132a4c64ca51ecd9ee0158c7f645e102f401ac64ca439

--- a/wire/peer_printgen.h
+++ b/wire/peer_printgen.h
@@ -74,4 +74,4 @@ void printwire_onion_message(const char *fieldname, const u8 *cursor);
 void printwire_channel_update_checksums(const char *fieldname, const u8 **cursor, size_t *plen);
 void printwire_channel_update_timestamps(const char *fieldname, const u8 **cursor, size_t *plen);
 #endif /* LIGHTNING_WIRE_PEER_PRINTGEN_H */
-// SHA256STAMP:b3a92a714208711191936bbdb7ff0f0a2ca412941ea95e67210029e5249f3d60
+// SHA256STAMP:1b0c5319cd9ab8c0281132a4c64ca51ecd9ee0158c7f645e102f401ac64ca439

--- a/wire/peer_wiregen.c
+++ b/wire/peer_wiregen.c
@@ -1749,4 +1749,4 @@ bool fromwire_channel_update_option_channel_htlc_max(const void *p, secp256k1_ec
  	*htlc_maximum_msat = fromwire_amount_msat(&cursor, &plen);
 	return cursor != NULL;
 }
-// SHA256STAMP:d664c34d8dddb0c11be08e907fb3f58deda009cee08661afa7030101c2a7c5f3
+// SHA256STAMP:1b0c5319cd9ab8c0281132a4c64ca51ecd9ee0158c7f645e102f401ac64ca439

--- a/wire/peer_wiregen.c
+++ b/wire/peer_wiregen.c
@@ -1749,4 +1749,4 @@ bool fromwire_channel_update_option_channel_htlc_max(const void *p, secp256k1_ec
  	*htlc_maximum_msat = fromwire_amount_msat(&cursor, &plen);
 	return cursor != NULL;
 }
-// SHA256STAMP:b3a92a714208711191936bbdb7ff0f0a2ca412941ea95e67210029e5249f3d60
+// SHA256STAMP:d664c34d8dddb0c11be08e907fb3f58deda009cee08661afa7030101c2a7c5f3

--- a/wire/peer_wiregen.h
+++ b/wire/peer_wiregen.h
@@ -650,4 +650,4 @@ bool fromwire_channel_update_option_channel_htlc_max(const void *p, secp256k1_ec
 
 
 #endif /* LIGHTNING_WIRE_PEER_WIREGEN_H */
-// SHA256STAMP:d664c34d8dddb0c11be08e907fb3f58deda009cee08661afa7030101c2a7c5f3
+// SHA256STAMP:1b0c5319cd9ab8c0281132a4c64ca51ecd9ee0158c7f645e102f401ac64ca439

--- a/wire/peer_wiregen.h
+++ b/wire/peer_wiregen.h
@@ -650,4 +650,4 @@ bool fromwire_channel_update_option_channel_htlc_max(const void *p, secp256k1_ec
 
 
 #endif /* LIGHTNING_WIRE_PEER_WIREGEN_H */
-// SHA256STAMP:b3a92a714208711191936bbdb7ff0f0a2ca412941ea95e67210029e5249f3d60
+// SHA256STAMP:d664c34d8dddb0c11be08e907fb3f58deda009cee08661afa7030101c2a7c5f3

